ouroboros-consensus-mock-test/test/Test/ThreadNet/Praos.hs

```diff
@@ -30,22 +32,44 @@ import           Test.ThreadNet.Util.NodeRestarts
 import           Test.ThreadNet.Util.NodeToNodeVersion
 import           Test.ThreadNet.Util.SimpleBlock
 
+import           Ouroboros.Consensus.Node.ProtocolInfo
+                     (NumCoreNodes (NumCoreNodes), enumCoreNodes)
 import           Test.Util.HardFork.Future (singleEraFuture)
 import           Test.Util.Orphans.Arbitrary ()
+import           Test.Util.Slots (NumSlots (unNumSlots))
 
 data TestSetup = TestSetup
-  { setupEpochSize    :: EpochSize
-  , setupInitialNonce :: Natural
+  { setupEpochSize     :: EpochSize
+  , setupInitialNonce  :: Natural
     -- ^ the initial Shelley 'praosInitialEta'
     --
     -- This test varies it too ensure it explores different leader schedules.
-  , setupK            :: SecurityParam
-  , setupNodeJoinPlan :: NodeJoinPlan
-  , setupSlotLength   :: SlotLength
-  , setupTestConfig   :: TestConfig
+  , setupK             :: SecurityParam
+  , setupNodeJoinPlan  :: NodeJoinPlan
+  , setupSlotLength    :: SlotLength
+  , setupTestConfig    :: TestConfig
+  , setupEvolvingStake :: PraosEvolvingStake
   }
   deriving (Show)
 
+genEvolvingStake :: EpochSize -> TestConfig -> Gen PraosEvolvingStake
+genEvolvingStake epochSize TestConfig {numSlots, numCoreNodes} = do
+    chosenEpochs <- sublistOf [1..EpochNo $ max 1 maxEpochs - 1]
```

Comment:
This list enumeration should start at `0` now, since `latestEvolvedStakeDistAsOfEpoch` (`Map.split` specifically) would drop a `(0, sd)` entry when applied at epoch `0`.

Reply:
Indeed! fixed.